076-connector-offsets-support.md

```diff
@@ -0,0 +1,286 @@
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to list the connector offsets due to missing property listOffsets in KafkaConnector CR.
+    reason: ListOffsets
+    status: "True"
+    type: Warning
+  ```
+  The operator will leave the annotation on the KafkaConnector resource until either the list operation succeeds, or the user removes the annotation.
+  This means the operator will retry the list on every reconciliation, allowing the condition to remain present for the user to see.
+
+### Altering offsets
+
+To alter offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connector-offsets` set to `alter`.
+After the annotation is added, on the next reconciliation the operator will call the PATCH /connectors/{connector}/offsets endpoint to alter the offsets.
+Once the patch is complete the operator will then remove the annotation from the KafkaConnector CR.
+
+The operator will read in the entry called `offsets.json` from the ConfigMap.
+The entry name matches the entry name used in the list operation, so that a user can list offsets, then edit the ConfigMap the operator created, then request the offsets be altered using that same ConfigMap.
+
+The name of the ConfigMap the operator will use to construct the request body will be set by the user in the KafkaConnector CR.
+The KafkaConnector CRD will be updated to contain a new property called `alterOffsets` that must be set for the `strimzi.io/connector-offsets=alter` annotation to take effect.
+The structure of the `alterOffsets` property will be as below, to allow it to be extended in future if required:
+
+```yaml
+alterOffsets:
+  fromConfigMap:
+    name: my-connector-offsets
+```
+
+Notes:
+* The format of the request body will be validated to make sure it is syntactically valid JSON, but the operator won't perform any further validation, instead relying on the API endpoint to return a reasonable error message.
+* If the request to the Connect API fails the operator will add a condition to the KafkaConnector CR status with a warning message that includes the response from the API, e.g.
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to alter the connector offsets due to "message from endpoint".
+    reason: AlterOffsets
+    status: "True"
+    type: Warning
+  ```
+  The operator will leave the annotation on the KafkaConnector resource until either the patch operation succeeds, or the user removes the annotation.
+  This means the operator will retry the patch on every reconciliation, allowing the condition to remain present for the user to see.
+* Strimzi will shortcut and automatically fail to alter the offsets if the KafkaConnector resource does not have it's `state` set as `stopped`.
```

Comment:
shortcut meaning ending the operation early?

Reply:
`paused` state not sufficient?

Reply:
> shortcut meaning ending the operation early?

Yes so rather than making the request to Kafka Connect and dealing with the error response back, we don't make the request at all and just inform the user of the error.

> `paused` state not sufficient?

My understanding is that `paused` is not sufficient, is that yours as well @mimaison?